onedal/decomposition/pca.py

```diff
@@ -15,30 +15,75 @@
 # ==============================================================================
 
 import numpy as np
+from sklearn.decomposition._pca import _infer_dimension
+from sklearn.utils.extmath import stable_cumsum
 
 from daal4py.sklearn._utils import sklearn_check_version
 
 from ..common._base import BaseEstimator
-from ..common.hyperparameters import get_hyperparameters
+from ..common._policy import _get_policy
 from ..datatypes import _convert_to_supported, from_table, to_table
 
 
 class PCA(BaseEstimator):
     def __init__(
-        self, n_components=None, is_deterministic=True, method="precomputed", copy=True
+        self,
+        n_components=None,
+        is_deterministic=True,
+        method="cov",
+        whiten=False,
     ):
         self.n_components = n_components
         self.method = method
         self.is_deterministic = is_deterministic
+        self.whiten = whiten
 
-    def get_onedal_params(self, data):
+    def get_onedal_params(self, data, stage=None):
+        if stage is None:
+            n_components = self._resolve_n_components_for_training(data.shape)
+        elif stage == "predict":
+            n_components = self.n_components_
         return {
             "fptype": "float" if data.dtype == np.float32 else "double",
             "method": self.method,
-            "n_components": self.n_components,
+            "n_components": n_components,
             "is_deterministic": self.is_deterministic,
+            "whiten": self.whiten,
         }
 
+    def _get_policy(self, queue, *data):
+        return _get_policy(queue, *data)
+
+    def _resolve_n_components_for_training(self, shape_tuple):
+        if self.n_components is None or self.n_components == "mle":
+            return min(shape_tuple)
+        elif 0 < self.n_components < 1:
```

Comment:
Looks weird. Please double check

Reply:
Looks like it will be evaluated as follows:
1) Will be evaluated the first comparison. The Boolean value will be returned
2) Boolean value will be compared with <1, equal to negating it.

Reply:
works fine :)